caffe2/core/predictor.cc

```diff
@@ -48,9 +48,11 @@ void Predictor::run(const TensorVector& inputs, TensorVector* outputs) {
 
   CAFFE_ENFORCE(ws_.RunNet(run_net_.name()));
 
-  outputs->resize(run_net_.external_output_size());
-  for (auto i = 0; i < outputs->size(); ++i) {
-    (*outputs)[i] = extractOutputTensor(&ws_, run_net_.external_output(i));
+  if (outputs != nullptr) {
```

Comment:
Yes, I am running from C++. Actually I am not using this predictor in my project now, but directly forwarding my net from the workspace, with inputs and outputs transferred using the "CopyFrom" API. I am using a network transferred from Caffe, and if I initialize it on GPU, I find it cannot be used from a predictor because every layer, including the "data" layer and the output layers, contain TensorCUDA, but the predictor enforces the IO blobs to contain TensorCPU.

Maybe I have some misunderstanding of the purpose of this predictor. If this modification conflicts with your design purpose, just omit it.